examples/gno.land/r/gov/dao/v3/memberstore/types.gno

```diff
@@ -0,0 +1,172 @@
+package memberstore
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+type ErrMemberAlreadyExists struct {
+	Tier string
+}
+
+func (e *ErrMemberAlreadyExists) Error() string {
+	return "member already exists on tier " + e.Tier
+}
+
+type Member struct {
+	InvitationPoints int
+}
+
+func (m *Member) RemoveInvitationPoint() {
+	if m.InvitationPoints <= 0 {
+		panic("not enough invitation points")
+	}
+
+	m.InvitationPoints = m.InvitationPoints - 1
+}
```

Comment:
It seems that `InvitationPoints` property could be private to make sure only `RemoveInvitationPoint()` is used to change its value

Reply:
Might be an option, but doing that won't allow us to easily create Members from outside the memberstorage package. Also, I should add extra methods to get how many invitation points are available. Do you see any corner case where we must make that field private?

Reply:
At this point I'm unsure if the `InvitationPoints` field can be modified in a place where it shouldn't 🤔, but if so that would be a corner case that would require that field to be private.

Other than that what I thought before was that the `Member` type could potentially be a "data" type if the field is public, like:
```go
type Member struct {
  InvitationPoints uint64
}
```

That way the type would make sure the minimum value is zero, so callers would just use for example:
```go
m := Member{3}
m.InvitationPoints--
```

Maybe removing the need of a decrease method.